crates/subspace-networking/examples/requests.rs

```diff
@@ -31,10 +33,26 @@ async fn main() {
             println!("Request handler for request");
             Some(ExampleResponse)
         })],
-        prometheus_metrics_server_address: Some("127.0.0.1:63000".parse().unwrap()),
+        metrics_registry: Some(Default::default()),
         ..Config::with_generated_keypair()
     };
-    let (node_1, mut node_runner_1) = subspace_networking::create(config_1).await.unwrap();
+    let (node_1, mut node_runner_1, metric_registry) =
+        subspace_networking::create(config_1).await.unwrap();
```

Comment:
Why does it return a registry? It is an input, so you already have it even before this function is called.

Reply:
It's not clonable.

Reply:
Hm... can you send a mutable reference to it in the config then?

Reply:
Also, you can pass None because you don't have an external registry and you'll get the default one with the configured metrics.

Reply:
Well, I'd expect to not pay for metrics if I didn't specify it, a bit unexpected (to me).
Let's not make metrics so special and central to the API. It should be just an optional thing, in case it is not provided, nothing should be happening.